composer/trainer/trainer.py

```diff
@@ -1943,6 +1943,16 @@ def __init__(
                     auto_microbatching,
                 )
 
+        # Set the iteration timestamp to the overall timestamp if loading from a checkpoint that was created before
+        # iteration was introduced in Composer v0.19.1. This is necessary to ensure that the iteration timestamp is
+        # accurate for callbacks and backwards compatibility for checkpoints.
+        if (
+            self.state.timestamp.iteration == 0 and self.state.timestamp.token_in_iteration == 0 and
+            self.state.timestamp.epoch_in_iteration == 0
```

Comment:
why does epcoh_in_iteration need to be 0?

Reply:
If the checkpoint is old, then both tokens and epoch should be 0